test/Operators/finitedifference/opt.jl

```diff
@@ -224,6 +224,9 @@ end
             center_values = ones(FT, center_space)
             center_velocities = Geometry.WVector.(center_values)
 
+            function_filter(@nospecialize(ft)) =
+                ft !== typeof(Base.CoreLogging.handle_message)
+
```

Comment:
This doesn't seem to be a great solution, as each `@test_opt` that fails this needs to have this function filter added. Plus, other tests still fail.

I'm a little surprised that we are seeing this `runtime dispatch detected: Base.CoreLogging.handle_message` since [`handle_message` uses `@nospecialize`](https://github.com/JuliaLang/julia/blob/244ada361432462012835c93d3bac031e8046793/base/logging.jl#L661-L663). At least given my current understanding of runtime dispatch and `@nospecialize`... 

@simonbyrne any tips?

Reply:
@aviatesk any idea what is causing the problem here?

Reply:
That `@nospecialize` doesn't eliminate dynamic dispatch when `logger` is abstract as `handle_message` has multiple methods for different loggers.

I'd say that usage of `function_filter` is a solution until we make the logging module type stable, which I'm not sure is worthwhile as I guess logging is designed to be very dynamic for some good reasons.